app/views/fields/polymorphic/_index.html.erb

```diff
@@ -18,7 +18,7 @@ By default, the relationship is rendered as a link to the associated object.
 
 <% if field.data %>
   <%= link_to(
-    field.data.to_s,
+    field.associated_resource,
```

Comment:
Or `displayed_associated_resource`, in reference to the above comment. Not sure which one is better.